### PR TITLE
repolist: Implement JSON output

### PR DESCRIPTION
**Pull request by jan-kolarik**
- Created shared `--json` option for requesting output in JSON format
- Implementation for `repo list` command

For: https://github.com/rpm-software-management/dnf5/issues/867.

### CODE REVIEW

**Comment by jan-kolarik** [comment]
Example output:
```
[root@71dd3fed55a5 dnf5]# dnf5 repolist --json
[
  {
    "id":"copr:copr.fedorainfracloud.org:rpmsoftwaremanagement:dnf-nightly",
    "name":"Copr repo for dnf-nightly owned by rpmsoftwaremanagement",
    "is_enabled":true
  },
  {
    "id":"copr:copr.fedorainfracloud.org:rpmsoftwaremanagement:test-utils",
    "name":"Copr repo for test-utils owned by rpmsoftwaremanagement",
    "is_enabled":true
  },
  {
    "id":"updates",
    "name":"Fedora 39 - x86_64 - Updates",
    "is_enabled":true
  },
  {
    "id":"fedora-cisco-openh264",
    "name":"Fedora 39 openh264 (From Cisco) - x86_64",
    "is_enabled":true
  },
  {
    "id":"fedora",
    "name":"Fedora 39 - x86_64",
    "is_enabled":true
  }
]
```

**Comment by kontura** [comment]
> The question is also if to have consistent attributes in the JSON output or if to filter some (like `is_enabled`) based on the user input.

Personally I prefer consistent attributes, I believe it will be less error prone.

File: dnf5/context.cpp [hunk 1]
```diff
@@ -206,6 +209,7 @@ class Context::Impl {
     const char * comment{nullptr};
 
     bool should_store_offline = false;
+    bool json_output = false;
```
[the remark this answers is not in the document]
Comment (jan-kolarik) [reply dnf5/context.cpp:212]: I've been already working on the follow-up `repoinfo` implementation and there I am calling the `set_quiet(true)` when json output is requested, so this place is handy in this regard.

[the remark this answers is not in the document]
Comment (kontura) [reply dnf5/context.cpp:212]: I see, however doesn't that have the same issue as mentioned in https://github.com/rpm-software-management/dnf5/pull/1524?
That is `dnf5 repo info --quiet --refresh` prints REPOSYNC to stdout which would ruin the json.

[the remark this answers is not in the document]
Comment (jan-kolarik) [reply dnf5/context.cpp:212]: Hmm, you're right this is happening, though I'd think it's suppressed by this line: https://github.com/rpm-software-management/dnf5/blob/5.2.3.0/dnf5/main.cpp#L1192. So it seems like a bug to me right now. I am still planning to do such binding to `quiet` there...

[the remark this answers is not in the document]
Comment (jan-kolarik) [reply dnf5/context.cpp:212]: So the issue is that the ^ mentioned callbacks setup should be done after parsing the command line arguments. Then the `quiet` is taken into account. Tested locally and seems working :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (kontura) [reply dnf5/context.cpp:212]: Thank you for looking into it.

**Pull request merged**